src/tag-editor/__tests__/get-tags-diff.test.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { warnOnce } from '@cloudscape-design/component-toolkit/internal';
+
+import { TagEditorProps } from '../../../lib/components/tag-editor';
+import { getTagsDiff } from '../utils';
+
+jest.mock('@cloudscape-design/component-toolkit/internal', () => ({
+  warnOnce: jest.fn(),
+}));
+
+describe('identifyTagStates', () => {
```

Comment:
To make sure we cover all corner cases, we should have "all tags created" test case `getTagsDiff({}, newTags)` and "all tags removed" – `getTagsDiff(initialTags, {})`

Reply:
good call!

added these tests

Reply:
Still do not see this test case: `getTagsDiff(initialTags, [])`